pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableConfigsRestletResource.java

```diff
@@ -322,26 +322,25 @@ public ConfigSuccessResponse updateConfig(
       @ApiParam(value = "Force update the table schema") @DefaultValue("false")
       @QueryParam("forceTableSchemaUpdate") boolean forceTableSchemaUpdate,
       String tableConfigsStr, @Context HttpHeaders headers) throws Exception {
+    tableName = DatabaseUtils.translateTableName(tableName, headers);
     Pair<TableConfigs, Map<String, Object>> tableConfigsAndUnrecognizedProps;
     TableConfigs tableConfigs;
     String translatedTableName;
     try {
-      translatedTableName = DatabaseUtils.translateTableName(tableName, headers);
       tableConfigsAndUnrecognizedProps =
           JsonUtils.stringToObjectAndUnrecognizedProperties(tableConfigsStr, TableConfigs.class);
       tableConfigs = tableConfigsAndUnrecognizedProps.getLeft();
-      Preconditions.checkState(
-          DatabaseUtils.translateTableName(tableConfigs.getTableName(), headers).equals(translatedTableName),
-          "Table name mismatch: %s is not equivalent to %s", tableConfigs.getTableName(), tableName);
+      tableConfigs.setTableName(DatabaseUtils.translateTableName(tableConfigs.getTableName(), headers));
```

Comment:
setting the table name here will make all the validations on table/schema names redundant which are part of `validateConfig(tableConfigs, typesToSkip);` called below

Reply:
Good point. Changed the order